app/controllers/users_controller.rb

```diff
@@ -1,82 +1,58 @@
 class UsersController < ApplicationController
-  before_action :set_user, only: [:show, :edit, :update, :destroy]
+  before_action only: [:show, :edit, :update]
 
-  # GET /users
-  # GET /users.json
-  def index
-    @users = User.all
-  end
-
-  # GET /users/1
-  # GET /users/1.json
   def show
     @user = User.find(params[:id])
   end
-
-  # GET /users/new
-  def new
-  end
-
+  
   # GET /users/1/edit
   def edit
+    @user = User.find(params[:id])
   end
 
-  # POST /users
-  # POST /users.json
-  def create
-    # if @user.name.empty?
-    #   flash[:notice]
-    @user = User.new(user_params)
-    if @user.email.empty? or ((@user.email.upcase =~ /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/) != 0)
-      flash[:error] = "Invalid email address."
-      render 'new'
-    elsif @user.password.length < 4
-      flash[:error] = "Your password must at least be 4 letters."
-      render 'new'
-    else
-      if @user.save
-        flash[:notice] = "Success!"
-        redirect_to @user
-      else
-        render 'new'
-      end
-    end
-  end
-  
-
   # PATCH/PUT /users/1
   # PATCH/PUT /users/1.json
   def update
-    respond_to do |format|
-      if @user.update(user_params)
-        format.html { redirect_to @user, notice: 'User was successfully updated.' }
-        format.json { render :show, status: :ok, location: @user }
+    if not params[:id]
+      redirect_to root_path
+    end
+    user = User.find(params[:id])
+    # update fields
+    user.first_name = user_params[:first_name]
+    user.last_name = user_params[:last_name]
+    user.age = user_params[:age]
+    user.education = user_params[:education]
+    user.school = user_params[:school]
+    user.expertise = user_params[:expertise]
+    user.description = user_params[:description]
+    user.availability = user_params[:availability]
+    user.save
+    
+    # update complete flag
+    if not user.complete
+      all_fields_filled = true
+      user_params.each do |k, v|
+        if v.blank?
+          all_fields_filled = false
+        end
+      end
+      if all_fields_filled 
+        user.complete = true
```

Comment:
You might want `user.complete` to be a method that's dynamically returned instead of a field.